pygfx/renderers/wgpu/engine/shared.py

```diff
@@ -130,6 +142,66 @@ def _create_diagnostics(self):
         PygfxResourceDiagnostics("pygfx_resources")
 
 
+def select_power_preference(power_preference):
+    """Select whether a powerful or battery-friendly GPU is selected.
+
+    Accepts a value from ``wgpu.PowerPreference``: "high-performance" or "low-power".
+
+    This function must be called before before the first ``Renderer`` is created.
+    """
+    if power_preference not in wgpu.PowerPreference:
+        raise ValueError(
+            "select_power_preference() received invalid value for {repr(wgpu.PowerPreference)}."
+        )
+    if Shared._instance is not None:
+        raise RuntimeError(
+            "The select_power_preference() function must be called before creating the first renderer."
+        )
+    Shared._power_preference = power_preference
+
+
+def select_adapter(adapter):
+    """Select a specific adapter / GPU.
+
+    Select an adapter as obtained via ``wgpu.gpu.enumerate_adapters()``, which
+    can be useful in multi-gpu environments.
+
+    For example::
+
+        adapters = wgpu.gpu.enumerate_adapters()
+        adapters_tesla = [a for a in adapters if "Tesla" in a.summary]
+        adapters_discrete = [a for a in adapters if "DiscreteGPU" in a.summary]
+        pygfx.renderes.wgpu.select_adapter(adapters_discrete[0])
+
+    Note that using this function reduces the portability of your code, because
+    it's highly specific for your current machine/environment.
+
+    The order of the adapters returned by ``wgpu.gpu.enumerate_adapters()`` is
+    such that Vulkan adapters go first, then Metal, then D3D12, then OpenGL.
+    Within each category, the order as provided by the particular backend is
+    maintained. Note that the same device may be present via multiple backends
+    (e.g. vulkan/opengl).
+
+    We cannot make guarantees about whether the order of the adapters matches
+    the order as reported by e.g. ``nvidia-smi``. We have found that on a Linux
+    multi-gpu cluster, the order does match, but we cannot promise that this is
+    always the case. If you want to make sure, do some testing by allocating big
```

Comment:
Maybe add the few lines to make a buffer to the docstring here?

Reply:
done